universal-application-tool-0.0.1/app/repository/ProgramRepository.java

```diff
@@ -159,40 +164,93 @@ public Program createOrUpdateDraft(Program existingProgram) {
     return database
         .find(Program.class)
         .where()
-        .eq(
-            "name",
-            database.find(Program.class).setId(programId).select("name").findSingleAttribute())
+        .in("name", database.find(Program.class).select("name").where().eq("id", programId).query())
+        .query()
         .findList()
         .stream()
         .collect(ImmutableList.toImmutableList());
   }
 
-  public ImmutableList<Long> getAllProgramVersionIds(long programId) {
+  public PaginationResult<Application> getApplicationsForAllProgramVersions(
```

Comment:
To summarize the below comments.  There's A LOT going on in the fuzzy matching that is undocumented (add a method javadoc) and doesn't all make sense for supporting to me.